File: app/src/main/java/tech/pauly/findapet/data/AnimalService.java
package tech.pauly.findapet.data;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Query;
import tech.pauly.findapet.data.models.AnimalListResponse;

public interface AnimalService {

    @GET("pet.find")
    Single<AnimalListResponse> fetchAnimals(@Query("location") String location,
                                            @Query("key") String key,
                                            @Query("animal") String type,
                                            @Query("offset") int offset,
                                            @Query("count") int count,
                                            @Query("sex") String sex);
}

File: app/src/main/java/tech/pauly/findapet/data/AnimalService.kt
package tech.pauly.findapet.data

import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Query
import tech.pauly.findapet.data.models.AnimalListResponse
import tech.pauly.findapet.data.models.SingleAnimalResponse

interface AnimalService {

    @GET("pet.find")
    fun fetchAnimals(@Query("location") location: String,
                     @Query("key") key: String,
                     @Query("animal") type: String,
                     @Query("offset") offset: Int,
                     @Query("count") count: Int,
                     @Query("sex") sex: String?,
                     @Query("age") age: String?,
                     @Query("size") size: String?,
                     @Query("breed") breed: String): Single<AnimalListResponse>

    @GET("pet.get")
    fun fetchAnimal(@Query("key") key: String,
                    @Query("id") id: String): Single<SingleAnimalResponse>
}
